basyx/aas/adapter/http.py

```diff
@@ -591,21 +605,38 @@ def _get_shells(self, request: Request) -> Iterator[model.AssetAdministrationShe
     def _get_shell(self, url_args: Dict) -> model.AssetAdministrationShell:
         return self._get_obj_ts(url_args["aas_id"], model.AssetAdministrationShell)
 
-    def _get_submodels(self, request: Request) -> Iterator[model.Submodel]:
-        submodels: Iterator[model.Submodel] = self._get_all_obj_of_type(model.Submodel)
+    def _get_submodels(self, request: Request) -> Tuple[Iterator[model.Submodel], int]:
+        limit = request.args.get('limit', type=int, default=10)
+        cursor = request.args.get('cursor', type=int, default=0)
+        submodels: List[model.Submodel] = list(self._get_all_obj_of_type(model.Submodel))
+        # Apply pagination
+        start_index = cursor
+        end_index = cursor + limit
+        paginated_submodels: Iterator[model.Submodel] = iter(submodels[start_index:end_index])
         id_short = request.args.get("idShort")
         if id_short is not None:
-            submodels = filter(lambda sm: sm.id_short == id_short, submodels)
+            paginated_submodels = filter(lambda sm: sm.id_short == id_short, paginated_submodels)
         semantic_id = request.args.get("semanticId")
         if semantic_id is not None:
-            spec_semantic_id = (HTTPApiDecoder.base64urljson
-                                (semantic_id, model.Reference, False))  # type: ignore[type-abstract]
-            submodels = filter(lambda sm: sm.semantic_id == spec_semantic_id, submodels)
-        return submodels
+            spec_semantic_id = HTTPApiDecoder.base64urljson(semantic_id, model.Reference, False)  # type: ignore
+            paginated_submodels = filter(lambda sm: sm.semantic_id == spec_semantic_id, paginated_submodels)
```

Comment:
I think filtering should be applied before slicing the submodels, such that x submodels are returned if `limit` is set to x.